test/unit/test_id_storage.c

```diff
@@ -0,0 +1,187 @@
+#include <r_util.h>
+#include "minunit.h"
+
+static bool test_r_id_storage_toomany(void) {
+	bool r;
+	ut32 id0, id1, id2;
+	RIDStorage *ids = r_id_storage_new (20, 22);
+	r = r_id_storage_add (ids, "aaa", &id0);
+	mu_assert ("id0", r);
+	r = r_id_storage_add (ids, "bbb", &id1);
+	mu_assert ("id1", r);
+	r = r_id_storage_add (ids, "ccc", &id2);
+	mu_assert ("id2", !r);
+	mu_end;
+}
+
+static bool test_r_id_storage_wrong(void) {
+	bool r;
+	ut32 id0, id1;
+	RIDStorage *ids = r_id_storage_new (20, 10);
+	r = r_id_storage_add (ids, "aaa", &id0);
+	mu_assert ("id0", !r);
+	mu_end;
+}
+
+static bool test_r_id_storage_prev_next_eq_0(void) {
+	// test if next reverts prev
+	bool r;
+	ut32 id0, id1, id2;
+	RIDStorage *ids = r_id_storage_new (1, 15);
+	r = r_id_storage_add (ids, "aaa", &id0);
+	mu_assert ("id0", r);
+	mu_assert ("id0", id0 == 1);
+	r = r_id_storage_add (ids, "bbb", &id1);
+	mu_assert ("id1", r);
+	mu_assert ("id1", id1 == 2);
+	r = r_id_storage_add (ids, "ccc", &id2);
+	mu_assert ("id2", r);
+	mu_assert ("id2", id2 == 3);
+	ut32 id = id1;
+	r_id_storage_get_prev (ids, &id);
+	r_id_storage_get_next (ids, &id);
+	r_id_storage_free (ids);
+	
+	mu_assert_eq (id, id1, "r_id_storage_{next/prev} reversal 0");
```

Comment:
what is the difference between this test and the ones at https://github.com/radareorg/radare2/pull/17652/files#diff-8dacb57a2e0ae9396828e3bdd407a6b77c779b6db28117f84c904eab2613ebc0R61-R68 and at https://github.com/radareorg/radare2/pull/17652/files#diff-8dacb57a2e0ae9396828e3bdd407a6b77c779b6db28117f84c904eab2613ebc0R84-R90 

It seems to me they are testing exactly the same thing.

Reply:
im testing the lower limit is 3 and not 0. so maybe the same code path but different behaviour